Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestRule.mm

```diff
@@ -721,61 +729,59 @@ - (void)removeInvalidResourceTypesForKey:(NSString *)ruleConditionKey
+        modifiedCondition[ruleConditionRequestDomainsKey] = nil;
+        modifiedCondition[ruleConditionExcludedRequestDomainsKey] = nil;
+        return modifiedCondition;
+    };
+
+    // We have to create one rule per request domain...
+    if (NSArray *requestDomains = _condition[ruleConditionRequestDomainsKey]) {
+        for (NSString *requestDomain in requestDomains) {
+            NSString *combinedRequestDomainAndURLFilter = [self _combineRequestDomain:requestDomain withURLFilter:_condition[declarativeNetRequestRuleConditionURLFilterKey]];
+            [convertedRules addObjectsFromArray:[self _convertRulesWithModifiedCondition:createModifiedConditionsForURLFilter(combinedRequestDomainAndURLFilter) webKitActionType:webKitActionType chromeActionType:chromeActionType]];
+        }
+    } else
+        [convertedRules addObjectsFromArray:[self _convertRulesWithModifiedCondition:_condition webKitActionType:webKitActionType chromeActionType:chromeActionType]];
+
+    // ...and we also have to create one ignore-previous-rules rule per excluded request domain
+    if (NSArray *excludedRequestDomains = _condition[ruleConditionExcludedRequestDomainsKey]) {
+        for (NSString *excludedRequestDomain in excludedRequestDomains)
+            [convertedRules addObjectsFromArray:[self _convertRulesWithModifiedCondition:createModifiedConditionsForURLFilter(excludedRequestDomain) webKitActionType:@"ignore-previous-rules" chromeActionType:chromeActionType]];
     }
 
-    [convertedRules addObjectsFromArray:[self _convertedRulesForWebKitActionType:webKitActionType chromeActionType:chromeActionType]];
+    return convertedRules;
+}
+
+- (NSArray<NSDictionary *> *)_convertRulesWithModifiedCondition:(NSDictionary *)condition webKitActionType:(NSString *)webKitActionType chromeActionType:(NSString *)chromeActionType
+{
+    NSMutableArray<NSDictionary<NSString *, id> *> *convertedRules = [NSMutableArray array];
+
+    if ([webKitActionType isEqualToString:@"make-https"])
+        [convertedRules addObject:[self _webKitRuleWithWebKitActionType:@"ignore-previous-rules" chromeActionType:chromeActionType condition:condition]];
 
-    if (_condition[ruleConditionInitiatorDomainsKey] && _condition[ruleConditionExcludedInitiatorDomainsKey]) {
+    [convertedRules addObject:[self _webKitRuleWithWebKitActionType:webKitActionType chromeActionType:chromeActionType condition:condition]];
+
+    if (condition[ruleConditionInitiatorDomainsKey] && condition[ruleConditionExcludedInitiatorDomainsKey]) {
         // If a rule specifies both initiatorDomains and excludedInitiatorDomains, we need to turn that into two rules. The first rule will have the excludedInitiatorDomains, and be implemented
         // as an ignore-previous-rules using if-frame-url (instead of unless-frame-url).
         // To do this, make a copy of the condition dictionary, make the initiatorDomains be the excludedInitiatorDomains of the original rule, and create an ignore-previous-rules rule.
```

Comment:
I think this comment is out of date, `ignore-previous-rules` comes second (I think).